### PR TITLE
Ct 1510 close a sar

### PR DESCRIPTION
**Pull request by beccapearce**

### CODE REVIEW

File: app/helpers/cases_helper.rb [hunk 1]
```diff
@@ -27,6 +27,11 @@ def action_button_for(event)
               respond_case_path(@case),
               id: 'action--mark-response-as-sent',
               class: 'button'
+    when :mark_response_sent_and_close
+      link_to t('common.case.mark_response_sent_and_close'),
+              respond_case_path(@case),
+              id: 'action--mark-response-as-sent-and-close',
+              class: 'button'
```
Comment (misaka) [inline app/helpers/cases_helper.rb]: I _want_ to say "AAHHH DON'T ADD ANYTHING TO THIS METHOD USE ACTION_LINK* BELOW AAAGGHHGHH!!!" but ... I kind of just want to see this get merged. So if you see me sighing morosely for a bit ... that's why.

File: app/models/concerns/case_states.rb [hunk 2]
```diff
@@ -81,11 +81,15 @@ def respond(current_user)
   end
 
   def close(current_user)
-    state_machine.close!(acting_user: current_user, acting_team: managing_team)
+    state_machine.close!(acting_user: current_user, acting_team: find_team(current_user))
```
Comment (misaka) [inline app/models/concerns/case_states.rb]: By using `find_team(current_user)` here you're anticipating the case may be closed by the responder, but that's only true for SAR cases, right? FOI cases shouldn't ever be closed by a responder.

File: app/decorators/case_transition_decorator.rb [hunk 3]
```diff
@@ -19,7 +19,12 @@ def event_and_detail
 
   private
   def event
-    Case::FOI::StandardStateMachine.event_name(object.event)
+    state_machine = object.case.state_machine
+    if state_machine.is_a?(ConfigurableStateMachine::Machine)
+      state_machine.event_name(object.event)
+    else
+      Case::FOI::StandardStateMachine.event_name(object.event)
```
Comment (misaka) [inline app/decorators/case_transition_decorator.rb]: Let's try to remember to add a `Case::FOI::StandardStateMachine#event_name` method that calls this class method `Case::FOI::StandardStateMachine.event_name` here. That way, we'll be able to get rid of all this code and just return `state_machine.event_name(object.event)` here.

Not required for this PR.

File: app/policies/case/sar_policy.rb [hunk 4]
```diff
@@ -14,17 +8,29 @@ def show?
...
       kase.responding_team.in? user.responding_teams
     end
   end
 
-
-  def new_case_link?
-    clear_failed_checks
-    check_can_trigger_event(:link_a_case) &&
-      check_user_is_a_manager_for_case
+  check :user_is_a_responder_for_case do
```
Comment (misaka) [inline app/policies/case/sar_policy.rb]: Should this be different from the [same check in `Case::BasePolicy`](https://github.com/ministryofjustice/correspondence_tool_staff/blob/1d3bc1540b92f5a6c17850afffbfd12d4c6251d8/app/policies/case/base_policy.rb#L465)?

**Review by stephenrichards — APPROVED** [review]

**Pull request merged**